### PR TITLE
Query-and-Update-Animals-Table

### PR DESCRIPTION
**Pull request by Kumilachew-g**
In this PR, I have done the following tasks.

- Added a column species of type string to animals table.
- Inserted the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

- Added the above insertions to data.sql file.
- Inside a transaction updated the animals table by setting the **species** column to **unspecified**. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:
   - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
   - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
   - Commit the transaction.
   - Verify that change was made and persists after commit.
   - Taken a screenshot of the results of your actions.
      
  image

- Taken a deep breath and... Inside a transaction deleted all records in the animals table, then roll back the transaction.
- After the rollback verifyied if all records in the animals table still exists. After that, I could start breathing as usual ;)

- **Take a screenshot of the results of your actions.**
  image

- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
- Take a screenshot of the results of your actions.  
  image
 
   image

- Wrote queries to answer the following questions:
   - How many animals are there?
   - How many animals have never tried to escape?
   - What is the average weight of animals?
   - Who escapes the most, neutered or not neutered animals?
   - What is the minimum and maximum weight of each type of animal?
   - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- Take a screenshot of the results of your queries.
    
image

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -14,3 +15,83 @@ SELECT * FROM animals WHERE neutered =true;
...
+   FROM animals 
+   GROUP BY neutered;
+   
+/*-- Find the max weight of animals --*/
+ 
+SELECT MAX(weight_kg) FROM animals;
+
+/*-- Find the min weight of animals --*/
+
+SELECT MIN(weight_kg) FROM animals;
```
Comment (gemmen29) [inline queries.sql]: - [ ] Kindly do those two query in one query and right implement it as they asked in this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=What%20is%20the%20minimum%20and%20maximum%20weight%20of%20each%20type%20of%20animal%3F) to get the minimum and maximum of each type of animal, not the minimum and maximum of animals table. 
### Hint: you need to use the `GROUP BY` statement.

File: queries.sql [hunk 2]
```diff
@@ -14,3 +15,83 @@ SELECT * FROM animals WHERE neutered =true;
...
+
+/*-- Calculate the average escape attemps of animals born between 1990 and 2000 --*/
+
+SELECT AVG(escape_attempts) FROM animals WHERE
+date_of_birth BETWEEN '01/01/1990' AND '01/01/2000'
```
Comment (gemmen29) [inline queries.sql]: - [ ] Kindly update this query as per this [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=What%20is%20the%20average%20number%20of%20escape%20attempts%20per%20animal%20type%20of%20those%20born%20between%201990%20and%202000%3F). You need to get the average number of escape attempts per animal type, not average escape attempts per animal table.

### Hint: you need to use the `GROUP BY` statement.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi <STUDENT_MENTION>, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯  
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kumilachew-g** [comment]
**Done based on review**
image

**Done based on review**

image

**Review by gemmen29 — APPROVED** [review]
Hi @Kumilachew-g, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**